shared/log.rs

```diff
@@ -32,6 +34,30 @@ pub fn get_log_file_handles(task_id: usize, settings: &Settings) -> Result<(File
     Ok((stdout, stderr))
 }
 
+fn command_data_to_text(mut data: &[u8]) -> Cow<str> {
+    if cfg!(windows) {
+        // On windows we run the command using "cmd" with a flag that makes it output Unicode (UTF16).
+
+        let is_odd = data.len() % 2 == 1;
+        let mut buffer = Vec::with_capacity(data.len() / 2 + if is_odd { 1 } else { 0 });
+
+        type CmdUtf16Endian = LittleEndian;
+
+        while data.len() > 1 {
+            buffer.push(data.read_u16::<CmdUtf16Endian>().unwrap());
+        }
+        if is_odd {
+            let extra = [*data.last().unwrap(), 0];
+            let mut extra = &extra as &[u8];
+            buffer.push(extra.read_u16::<CmdUtf16Endian>().unwrap());
+        }
+
+        Cow::from(String::from_utf16_lossy(&buffer))
```

Comment:
Since the Cow is reverted to a String anyway after being returned, just return a String directly instead of `Cow<str>`.
This removes a little bit of unnecessary complexity.